python/ray/serve/tests/test_standalone2.py

```diff
@@ -232,6 +232,47 @@ def f(*args):
     ray.shutdown()
 
 
+def test_no_controller_deserialization(start_and_shutdown_ray_cli_function):
+    """Ensure controller doesn't deserialize deployment_def or init_args/kwargs."""
+
+    runtime_env = {
+        "working_dir": (
+            "https://github.com/shrekris-anyscale/test_dag/archive/"
+            "1037818961158ccc754684fbf8b86cc2fe9ca040.zip"
```

Comment:
we really really need a local http version. (no one can push to your test_dag repo for example)

Reply:
actually can this test just use local working dir?

Reply:
> actually can this test just use local working dir?

Tasks can't take in [a local working dir](https://docs.ray.io/en/releases-1.12.1/ray-core/handling-dependencies.html#api-reference), so we would have to set the runtime_env in `ray.init()`. However, that runtime_env would also be shared by the Serve controller, so it would be able to deserialize this code, and the test would never fail.

> we really really need a local http version. (no one can push to your test_dag repo for example)

I think one potential solution is to add a new repo to the Ray project solely for these files. The Ray codebase wouldn't have the files locally, so they can be used to check runtime_env behavior. It would also allow all Ray contributors to make changes.

Reply:
You can put it in test_standalone and do:
- ray start cli (there's a fixture for this)
- ray.init("auto"), serve.start(detached=True), ray.shutdown
- ray.init("auto", working_dir=...).
- carry on

Reply:
Good idea! I made `test_controller_deserialization_deployment_def` use a local version of the graph, so it doesn't rely on the `test_dag` repo.